CHANGELOG.md

```diff
@@ -4,6 +4,14 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [unreleased]
 
+- Feature: Update `Marger` component. You can now pass an
+  object to the `top` and `bottom` props to handle breakpoints with `Marger`.
+  This update doesn't affect the previous props' validation, you can still pass
+  string values:
+  ```js
+  const topMarginsWithBreakpoints = { xs: 12, m: 2, default: 4 }
```

Comment:
Je trouve que ce n'est pas clair la notion de `orLess`. Dans l'exemple, en `s` je vais avoir `2` et `12` en `xxs`. C'est le `default` qui me met le doute en me disant que si je remplis pas la valeur , je vais avoir le `default`.

Reply:
Très bonne remarque, je vais réfléchir à mieux expliciter les ranges dans l'objet de props.

Reply:
Peut-être qu'il faut faire comme pour le `GridCol` et que `default` devienne `l` ?

Reply:
J'ai l'impression qu'on peut se passer du `default` si on décide d'avoir une stratégie "or-more" ou "or-less" vu que nos breakpoints `xxs` et `xxl` sont des ranges qui n'ont pas de limite (respectivement haute et basse).

Du coup je vois trois stratégies qu'on pourrait adopter :

**or-more**

| breakpoints   | xxs | xs |  s |  m | l  | xl | xxl |
|---------------|-----|----|----|----|----|----|-----|
| l: 4          |     |    |    |    |  4 |  4 |   4 |
| xxs: 2        |   2 |  2 |  2 |  2 |  2 |  2 |   2 |
| xl: 8         |     |    |    |    |    |  8 |   8 |
| xxs: 2, l: 4  |   2 |  2 |  2 |  2 |  4 |  4 |   4 |
| s: 1, xl: 4   |     |    |  1 |  1 |  1 |  4 |   4 |
| xxl: 9        |     |    |    |    |    |    |   9 |

**or-less**

| breakpoints   | xxs | xs |  s |  m | l  | xl | xxl |
|---------------|-----|----|----|----|----|----|-----|
| l: 4          |   4 |  4 |  4 |  4 |  4 |    |     |
| xxs: 2        |   2 |    |    |    |    |    |     |
| xl: 8         |   8 |  8 |  8 |  8 |  8 |  8 |     |
| xxs: 2, l: 4  |   2 |  4 |  4 |  4 |  4 |    |     |
| s: 1, xl: 4   |   1 |  1 |  1 |  4 |  4 |  4 |     |
| xxl: 9        |   9 |  9 |  9 |  9 |  9 |  9 |   9 |

**spécifiques**

| breakpoints   | xxs | xs |  s |  m | l  | xl | xxl |
|---------------|-----|----|----|----|----|----|-----|
| l: 4          |     |    |    |    |  4 |    |     |
| xxs: 2        |   2 |    |    |    |    |    |     |
| xl: 8         |     |    |    |    |    |  8 |     |
| xxs: 2, l: 4  |   2 |    |    |    |  4 |    |     |
| s: 1, xl: 4   |   1 |    |    |    |    |  4 |     |
| xxl: 9        |     |    |    |    |    |    |   9 |

Je crois que je préfère "or more" qui correspond plus à l'idée du "mobile first" qu'on a sur GridCol.

Reply:
@FlorentD @sunny @LanF3usT @camilledel @Morgan-jarry Est-ce que pour vous ça suffit du coup, le "or more" ? Ce qui transformerait : 
```js
<Marger top={viewportIsSOrLess ? 7 : 10}>
```
en 
```js
<Marger top={{
  xxs: 7,
  m: 10,
}} />
```

Personnellement, j'ai beaucoup de mal avec cette écriture, même du temps de bootstrap. C'est difficile pour moi de comprendre qu'un breakpoint signifie un range. Mais si on est tous d'accord, je change les queries et je pars là-dessus.

Reply:
je plussoie pour le `or more` qui me semble plus logique !

Reply:
Dans ton exemple, c'est `s: 7` plutôt ?

Reply:
C'est pour ça que je ne trouve pas ça très clair de mon côté.

Reply:
Un `<Marger top="7" top-m="10">` me parait plus lisible.

Reply:
D'un point de vue intégration, je pense que cette syntaxe est clair et logique. On a toujours tendance à partir du `xxs` pour aller vers un format desktop.

Reply:
Donc, je reprends la logique de `GridCol` et je l'intègre dans `Marger`, ça vous semble bien ? Comme ça, on a des composants de layout qui restent cohérent ? 

(au pire, si ce n'est pas vraiment clair à l'usage, on rechange).

Reply:
J'ai l'impression que c'est un bon compromis pour avancer sur le sujet.

Reply:
Je suis pas sûr que ce soit plus lisible `<Marger top="7" top-m="10">` . T'auras toujours le moment ou tu te demandes à quoi correspond `top="7"` . Après c'est vrai ça alourdit le code.

Reply:
Ok, testons.

Reply:
Alors personnellement, j'ai toujours eu du mal avec le `col="10" col-l="4"` par exemple. Parce que j'ai du mal à visualiser des breakpoints en range de viewports. Mais je pense que c'est bien si on part sur la même chose que `GridCol` en premier, histoire de rester cohérent. Quitte à rechanger plus tard après usage et discussion.

Reply:
Autre stratégie, qui a le mérite d'être plus claire, serait de faire des ranges où on précise les bornes. C'est plus verbeux mais c'est plus clair car on a pas d'ordre à connaître comme sur "or more" ou "or less".

**ranges**

| breakpoints            | xxs | xs |  s |  m | l  | xl | xxl |
|------------------------|-----|----|----|----|----|----|-----|
| xxsToL: 2              |   2 |  2 |  2 |  2 |  2 |    |     |
| xlToXxl: 8             |     |    |    |    |    |  8 |   8 |
| xxsToL: 2, xlToXxL: 4  |   2 |  2 |  2 |  2 |  4 |  4 |   4 |
| sToXl: 1, xxl: 4       |     |    |  1 |  1 |  1 |  1 |   4 |
| xs: 3, l: 4            |     |  3 |    |    |  4 |    |     |
| xl: 9, default: 1      |   1 |  1 |  1 |  1 |  1 |  1 |   9 |


Et du coup si on précise des ranges, ce n'est pas gênant d'avoir un `default`.


| breakpoints            | xxs | xs |  s |  m | l  | xl | xxl |
|------------------------|-----|----|----|----|----|----|-----|
| xxsToM: 1, lToXxL: 9   |   1 |  1 |  1 |  1 |  9 |  9 |   9 |
| default: 1, lToXxL: 9  |   1 |  1 |  1 |  1 |  9 |  9 |   9 |

Reply:
La logique de GridCol c'est celle de sunny ?

Reply:
Ca fait beaucoup de cas avec la version `To`.

Reply:
Oh, `fromXxs`, je n'y avais pas pensé. C'est pas mal pour indiquer un range, j'aime beaucoup. En terme d'écriture, ça ressemble à ce que `GridCol` fait.

Reply:
J'ai l'impression, en lisant le code sur KissKiss, que l'usage, c'est d'avoir une valeur par défaut et une autre à partir d'un breakpoint. Avoir un tableau de range me semble être fastidieux à écrire pour les cas d'usage qui existent.

Reply:
Avec l'idée de `from` + `orMore`, on aurait : 
```
<Marger top={ {
  default: 7,
  fromM: 10,
} } />
```

Reply:
Ah je viens de voir ton commentaire @sunny ! L'avantage de l'écriture avec le `To`, c'est que l'usage de `orMore` n'est pas implicite. Mais c'est un choix.

Reply:
Je crois que je préfère le `from` qui correspond beaucoup à nos cas d'usage, même si le `orMore` est implicite dans ce cas.

Reply:
`<Marger topXsToL="2" topXlToXxL="4" />` c'est hyper complexe comme écriture et dur à lire. J'aime bien l'idée de tester autre chose à le `from`.

Reply:
Le `from*` n'est pas aussi explicite qu'un range mais il est plus clair !

Et il n'interdit pas d'introduire des ranges plus tard (`fromL: 3, xxsToXs: 1`), ou de pouvoir mettre des valeurs uniques à côté (`fromL: 3, xxs: 1`) ou d'ajouter un `default` (équivalent à `fromXxs`).

| breakpoints                               | xxs | xs |  s |  m | l  | xl | xxl |
|-------------------------------------------|-----|----|----|----|----|----|-----|
| `<Marger top={{fromL: 4}} />`             |     |    |    |    |  4 |  4 |   4 |
| `<Marger top={{fromXxs: 2, fromL: 4}} />` |   2 |  2 |  2 |  2 |  4 |  4 |   4 |
| `<Marger top={{fromS: 1, fromXl: 4}} />`  |     |    |  1 |  1 |  1 |  4 |   4 |
| `<Marger top={{fromXxl: 9}} />`           |     |    |    |    |    |    |   9 |
| `<Marger top={{fromXxs: 1, fromL: 9}} />` |   1 |  1 |  1 |  1 |  9 |  9 |   9 |
| `<Marger top={{default: 1, fromL: 9}} />` |   1 |  1 |  1 |  1 |  9 |  9 |   9 |